firmware/boron-ins-fsm/src/imDoorSensor.cpp

```diff
@@ -18,6 +18,7 @@ bool doorTamperedFlag = false;
 bool doorMessageReceivedFlag = false;
 unsigned long doorHeartbeatReceived = 0;
 unsigned long doorLastMessage = 0;
+unsigned long timeWhenDoorClosed = 0;
```

Comment:
Can you test what happens when you restart the Sensor while the door sensor is closed?

I expect the first debug messages will have the 0x99 value.
Then after the first heartbeat, it will realize that the door is closed.
Does this realization reset the `timeWhenDoorClosed` or not? Do we want it to or not?

Reply:
0x99 is 0b10011001 which is "closed + heartbeat". In the new logic I am implementing, a transition from "closed heartbeat" to "closed heartbeat" would not reset the timer. However, the timer does still reset when the sensor resets since `millis()` resets to 0 as well. so if the realization happens when the occupation timer has not been exceeded it can enter state 1. I believe we don't want this.

Reply:
Just to have it written down... I think we chatted about this on Friday. I think that when we talked it through we decided that resetting to `millis()` is fine here. You're right that it does mean that we'll maybe get into state 1 when we didn't want to, but that's a better default than not ending up in state 1 when we should in the case of a restart during the daytime hours.